contrib/Joshua/scripts/localClusterStart.sh

```diff
@@ -161,27 +175,60 @@ function createClusterFile {
 	return ${status}
 }
 
+# Stop the Cluster from running.
+function stopCluster {
+	# Add an audit entree, if enabled
+	if [ "${AUDITCLUSTER}" -gt 0 ]; then
+		printf '%-15s (%6s)  Stopping cluster %-20s (%6s): %s\n' "$(date +'%Y-%m-%d %H:%M:%S')" "${$}" "${CLUSTERSTRING}" "${FDBSERVERID}" >> "${AUDITLOG}"
+	fi
+	if [ -z "${FDBSERVERID}" ]; then
+		log 'FDB Server process is not defined'
+		let status="${status} + 1"
+	elif ! kill -0 "${FDBSERVERID}"; then
+		log "Failed to locate FDB Server process (${FDBSERVERID})"
+		let status="${status} + 1"
+	elif "${BINDIR}/fdbcli" -C "${FDBCONF}" --exec "kill; kill ${CLUSTERSTRING}; sleep 3" --timeout 120 &>> "${LOGDIR}/fdbcli-kill.log"
+	then
+		log "Killed cluster (${FDBSERVERID}) via cli"
+
+	elif ! kill -9 "${FDBSERVERID}"; then
+		log "Failed to forcibly kill FDB Server process (${FDBSERVERID})"
+		let status="${status} + 1"
+	else
+		log "Forcibly killed FDB Server process (${FDBSERVERID})"
+	fi
+	return "${status}"
+}
+
 # Start the server running.
 function startFdbServer {
+	# Add an audit entree, if enabled
+	if [ "${AUDITCLUSTER}" -gt 0 ]; then
+		printf '%-15s (%6s)  Starting cluster %-20s\n' "$(date +'%Y-%m-%d %H:%M:%S')" "${$}" "${CLUSTERSTRING}" >> "${AUDITLOG}"
+	fi
+
 	if [ "${status}" -ne 0 ]; then
 		:
 	elif ! displayMessage "Starting Fdb Server"
 	then
 		log 'Failed to display user message'
 		let status="${status} + 1"
 
-	elif ! "${BINDIR}/fdbserver" -C "${FDBCONF}" -p "auto:${FDBSERVERPORT}" -L "${LOGDIR}" -d "${WORKDIR}/fdb/$$" &> "${LOGDIR}/fdbserver.log" &
+	elif ! "${BINDIR}/fdbserver" --knob_disable_posix_kernel_aio=1 -C "${FDBCONF}" -p "${IPADDRESS}:${FDBSERVERPORT}" -L "${LOGDIR}" -d "${WORKDIR}/fdb/${$}" &> "${LOGDIR}/fdbserver.log" &
 	then
 		log "Failed to start FDB Server"
 		# Maybe the server is already running
-		FDBSERVERID="$(pidof fdbserver)"
+		#FDBSERVERID="$(pidof fdbserver)"
```

Comment:
With the rest of these changes, should we just remove these commented lines as obsolete?

Reply:
done